pkg/networking/networking.go

```diff
@@ -87,7 +95,20 @@ func (n *networkImpl) AddHeaderField(key, value string) {
 
 func (n *networkImpl) AddHeaders(request *http.Request) error {
 	n.addDefaultHeader(request)
-	return middleware.AddAuthenticationHeader(n.GetAuthenticator(), n.config, request)
+	hc := &middleware.HeaderCapture{}
+	var rt http.RoundTripper = hc
+	rt = n.addMiddlewaresToRoundTripper(rt)
+	_, err := rt.RoundTrip(request)
+	for k, v := range hc.CapturedHeaders {
+		request.Header.Set(k, v)
```

Comment:
how does this work? `hc` is just initiated in the same method empty. I think it would be more explicit to name the `HeaderCapture` like `HeaderCapturingRoundtripper`, as I found the whole method implementation a bit confusing how about something like:

```
	rt, ok := n.addMiddlewaresToRoundTripper(&middleware.HeaderCapture{}).(CapturedHeaders)
	if !ok { return errors.New...) }
	_, err := rt.RoundTrip(request)
	for k, v := range rt.CapturedHeaders {
		request.Header.Set(k, v)
```

Reply:
Renamed 👍 

addMiddlewares... returns a different RoundTriper, so it can't be cast to CaptureHeaders (A -> B -> C -> CaptureHeadersMiddleware).